apps/settings/js/panels/operator_settings/models/network_type_manager.js

```diff
@@ -0,0 +1,88 @@
+/* global getSupportedNetworkInfo */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  var ObservableArray = require('modules/mvvm/observable_array');
+  var SettingsHelper = require('shared/settings_helper');
+
+  const SETTINGS_KEY = 'ril.radio.preferredNetworkType';
+
+  var NetworkTypeManager = Module.create(function NetworkTypeManager(conn) {
+    if (!conn.setPreferredNetworkType) {
+      this.throw(`mobile connection does not support settings preferred network
```

Comment:
The string was more than one line so I decided to use template string to save writing `+`.
And we don't have `setPreferredNetworkType` on CDMA devices. Although we already check it in the panel but I would like to make it throw an exception when it is misused.